protocol.go

```diff
@@ -0,0 +1,80 @@
+package tarantool
+
+import (
+	"fmt"
+)
+
+type ProtocolVersion uint64
+type Feature uint64
+
+type idInfo struct {
+	version  ProtocolVersion
+	features []Feature
+}
+
+const NoProtocolVersion ProtocolVersion = 0
+
+const (
+	// Streams support.
+	StreamsFeature Feature = 0
+	// Interactive transactions support.
+	TransactionsFeature Feature = 1
+	// Support of MP_ERROR objects over MessagePack.
+	ErrorExtensionFeature Feature = 2
+	// Support of watchers.
+	WatchersFeature Feature = 3
+)
+
+var featureName = map[Feature]string{
+	StreamsFeature:        "StreamsFeature",
+	TransactionsFeature:   "TransactionsFeature",
+	ErrorExtensionFeature: "ErrorExtensionFeature",
+	WatchersFeature:       "WatchersFeature",
+}
```

Comment:
It will be more universal to add String() method for Feature type.

```Go
func (f Feature) String() string {
    switch f {
    StreamsFeature:
        return "StreamsFeature"
    ...
    default:
        return fmt.Sprintf("unknown feature %d", f)
    }
}
```

Reply:
Thank you, fixed